src/time.rs

```diff
@@ -0,0 +1,136 @@
+//! Time structures.
+//!
+//! The `time` modules contains structures used to represent both
+//! absolute and relative time.
+
+use core::{ops, fmt};
+
+static MILLIS_PER_SEC: u64 = 1_000;
+
+/// A representation of an absolute time value.
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TimeStamp {
+    millis: u64,
+}
+
+impl TimeStamp {
+    /// Create a new `TimeStamp`.
+    pub fn new(millis: u64) -> TimeStamp {
+        TimeStamp { millis }
+    }
+
+    /// Add a [TimeInterval] to the current timestamp value and return a
+    /// `Option` containing a new [TimeStamp] of the result on success.
+    ///
+    /// [TimeInterval]: struct.TimeInterval.html
+    /// [TimeStamp]: struct.TimeStamp.html
+    pub fn advance(&self, rhs: TimeInterval) -> Option<TimeStamp> {
+        self.millis.checked_add(rhs.millis()).map(|x| TimeStamp::new(x))
```

Comment:
```
> 2**64 milliseconds -> million years
approx. 584.5545 million tropicalyear (time)
```
Really? Caring about overflows here?

Reply:
Renamed `advance` to `checked_add` so anyone that needs to worry about overflows can (no idea who will).